Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift

```diff
@@ -304,3 +346,56 @@ private func adjustSwiftCompilerArgumentsForIndexStoreUpdate(
   ]
   return result
 }
+
+/// Adjust compiler arguments that were created for building to compiler arguments that should be used for indexing.
+///
+/// This removes compiler arguments that produce output files and adds arguments to index the file.
+private func adjustClangCompilerArgumentsForIndexStoreUpdate(
+  _ compilerArguments: [String],
+  fileToIndex: DocumentURI
+) -> [String] {
+  let removeFlags: Set<String> = [
+    // Disable writing of a depfile
+    "-M",
+    "-MD",
+    "-MMD",
+    "-MG",
+    "-MM",
+    "-MV",
+    // Don't create phony targets
+    "-MP",
+    // Don't writ out compilation databases
+    "-MJ",
+    // Continue in the presence of errors during indexing
+    "-fmodules-validate-once-per-build-session",
+    // Don't compile
+    "-c",
+  ]
+
+  let removeArguments: Set<String> = [
+    // Disable writing of a depfile
+    "-MT",
+    "-MF",
+    "-MQ",
+    // Don't write serialized diagnostic files
+    "--serialize-diagnostics",
+  ]
+
+  var result: [String] = []
+  result.reserveCapacity(compilerArguments.count)
+  var iterator = compilerArguments.makeIterator()
+  while let argument = iterator.next() {
+    if removeFlags.contains(argument) || argument.starts(with: "-fbuild-session-file=") {
```

Comment:
Does `serailize-diagnostics` also allow `=`? Does `-fbuild-session-file` allow a separate arg rather than `=`?

Reply:
So, `serailize-diagnostics` does not allow `=` and `-fbuild-session-file` requires a `=`. But eg. `-MT` doesn’t require a space. And `serialize-diagnostics` can be spelled with a single or multiple spaces. 

I ended up generalizing the entire command line argument matching in https://github.com/apple/sourcekit-lsp/pull/1314